importar bibliotecas que sejam precisas -> vocês vão ter que ter as bibliotecas instaladas ou no anaconda e depois usar aqui ou instalar mesmo no vosso pc, mas depois podemos ver isto juntas

In [5]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

carregar o dataset (que tem de estar na mesma pasta que este notebook)

In [3]:
df = pd.read_csv('online_shoppers_intention.csv')
df.head()

,Administrative,Administrative_Duration,Informational,Informational_Duration,ProductRelated,ProductRelated_Duration,BounceRates,ExitRates,PageValues,SpecialDay,Month,OperatingSystems,Browser,Region,TrafficType,VisitorType,Weekend,Revenue
0,0,0.0,0,0.0,1,0.000000,0.20,0.20,0.0,0.0,Feb,1,1,1,1,Returning_Visitor,False,False
1,0,0.0,0,0.0,2,64.000000,0.00,0.10,0.0,0.0,Feb,2,2,1,2,Returning_Visitor,False,False
2,0,0.0,0,0.0,1,0.000000,0.20,0.20,0.0,0.0,Feb,4,1,9,3,Returning_Visitor,False,False
3,0,0.0,0,0.0,2,2.666667,0.05,0.14,0.0,0.0,Feb,3,2,2,4,Returning_Visitor,False,False
4,0,0.0,0,0.0,10,627.500000,0.02,0.05,0.0,0.0,Feb,3,3,1,4,Returning_Visitor,True,False


ver info toda sobre o dataset (parte 1)

In [2]:
print(f"Número de colunas: {df.shape[1]}")
print(f"Número de linhas: {df.shape[0]}")
print("\nColunas e tipos de dados:\n")
print(df.dtypes)

print("\n" + "="*60)
print("Contagem de valores por coluna")
print("="*60)

for coluna in df.columns:
    print(f"\n--- {coluna} ---")
    print(f"Valores únicos: {df[coluna].nunique()}")
    print(df[coluna].value_counts(dropna=False))

Número de colunas: 18
Número de linhas: 12330

Colunas e tipos de dados:

Administrative               int64
Administrative_Duration    float64
Informational                int64
Informational_Duration     float64
ProductRelated               int64
ProductRelated_Duration    float64
BounceRates                float64
ExitRates                  float64
PageValues                 float64
SpecialDay                 float64
Month                       object
OperatingSystems             int64
Browser                      int64
Region                       int64
TrafficType                  int64
VisitorType                 object
Weekend                       bool
Revenue                       bool
dtype: object

Contagem de valores por coluna

--- Administrative ---
Valores únicos: 27
Administrative
0     5768
1     1354
2     1114
3      915
4      765
5      575
6      432
7      338
8      287
9      225
10     153
11     105
12      86
13      56
14      44
15      38
16      24
17   

ver info toda sobre o dataset (parte 2)

In [3]:
from IPython.display import display

print(f"Número de colunas: {df.shape[1]}")
print(f"Número de linhas: {df.shape[0]}\n")

for coluna in df.columns:
    print(f"--- {coluna} (tipo: {df[coluna].dtype}) ---")
    
    tabela = df[coluna].value_counts(dropna=False).reset_index()
    tabela.columns = ['valor', 'contagem']
    tabela['percentagem'] = (tabela['contagem'] / len(df) * 100).round(2)
    
    display(tabela)
    print()

Número de colunas: 18
Número de linhas: 12330

--- Administrative (tipo: int64) ---


,valor,contagem,percentagem
0,0,5768,46.78
1,1,1354,10.98
2,2,1114,9.03
3,3,915,7.42
4,4,765,6.20
5,5,575,4.66
6,6,432,3.50
7,7,338,2.74
8,8,287,2.33
9,9,225,1.82



--- Administrative_Duration (tipo: float64) ---


,valor,contagem,percentagem
0,0.000000,5903,47.88
1,4.000000,56,0.45
2,5.000000,53,0.43
3,7.000000,45,0.36
4,11.000000,42,0.34
...,...,...,...
3330,65.600000,1,0.01
3331,552.200000,1,0.01
3332,206.250000,1,0.01
3333,401.150000,1,0.01



--- Informational (tipo: int64) ---


,valor,contagem,percentagem
0,0,9699,78.66
1,1,1041,8.44
2,2,728,5.90
3,3,380,3.08
4,4,222,1.80
5,5,99,0.80
6,6,78,0.63
7,7,36,0.29
8,9,15,0.12
9,8,14,0.11



--- Informational_Duration (tipo: float64) ---


,valor,contagem,percentagem
0,0.000000,9925,80.49
1,9.000000,33,0.27
2,10.000000,26,0.21
3,7.000000,26,0.21
4,6.000000,26,0.21
...,...,...,...
1253,1009.733333,1,0.01
1254,216.566667,1,0.01
1255,63.100000,1,0.01
1256,43.200000,1,0.01



--- ProductRelated (tipo: int64) ---


,valor,contagem,percentagem
0,1,622,5.04
1,2,465,3.77
2,3,458,3.71
3,4,404,3.28
4,6,396,3.21
...,...,...,...
306,378,1,0.01
307,254,1,0.01
308,183,1,0.01
309,210,1,0.01



--- ProductRelated_Duration (tipo: float64) ---


,valor,contagem,percentagem
0,0.000000,755,6.12
1,17.000000,21,0.17
2,11.000000,17,0.14
3,8.000000,17,0.14
4,15.000000,16,0.13
...,...,...,...
9546,4018.450000,1,0.01
9547,262.900000,1,0.01
9548,4627.489571,1,0.01
9549,11431.001240,1,0.01



--- BounceRates (tipo: float64) ---


,valor,contagem,percentagem
0,0.000000,5518,44.75
1,0.200000,700,5.68
2,0.066667,134,1.09
3,0.028571,115,0.93
4,0.050000,113,0.92
...,...,...,...
1867,0.018280,1,0.01
1868,0.013620,1,0.01
1869,0.000170,1,0.01
1870,0.028595,1,0.01



--- ExitRates (tipo: float64) ---


,valor,contagem,percentagem
0,0.200000,710,5.76
1,0.100000,338,2.74
2,0.050000,329,2.67
3,0.033333,291,2.36
4,0.066667,267,2.17
...,...,...,...
4772,0.008745,1,0.01
4773,0.013132,1,0.01
4774,0.002260,1,0.01
4775,0.020139,1,0.01



--- PageValues (tipo: float64) ---


,valor,contagem,percentagem
0,0.000000,9600,77.86
1,53.988000,6,0.05
2,42.293068,3,0.02
3,34.039975,2,0.02
4,54.980000,2,0.02
...,...,...,...
2699,6.094324,1,0.01
2700,28.253955,1,0.01
2701,16.090650,1,0.01
2702,24.897000,1,0.01



--- SpecialDay (tipo: float64) ---


,valor,contagem,percentagem
0,0.0,11079,89.85
1,0.6,351,2.85
2,0.8,325,2.64
3,0.4,243,1.97
4,0.2,178,1.44
5,1.0,154,1.25



--- Month (tipo: object) ---


,valor,contagem,percentagem
0,May,3364,27.28
1,Nov,2998,24.31
2,Mar,1907,15.47
3,Dec,1727,14.01
4,Oct,549,4.45
5,Sep,448,3.63
6,Aug,433,3.51
7,Jul,432,3.50
8,June,288,2.34
9,Feb,184,1.49



--- OperatingSystems (tipo: int64) ---


,valor,contagem,percentagem
0,2,6601,53.54
1,1,2585,20.97
2,3,2555,20.72
3,4,478,3.88
4,8,79,0.64
5,6,19,0.15
6,7,7,0.06
7,5,6,0.05



--- Browser (tipo: int64) ---


,valor,contagem,percentagem
0,2,7961,64.57
1,1,2462,19.97
2,4,736,5.97
3,5,467,3.79
4,6,174,1.41
5,10,163,1.32
6,8,135,1.09
7,3,105,0.85
8,13,61,0.49
9,7,49,0.40



--- Region (tipo: int64) ---


,valor,contagem,percentagem
0,1,4780,38.77
1,3,2403,19.49
2,4,1182,9.59
3,2,1136,9.21
4,6,805,6.53
5,7,761,6.17
6,9,511,4.14
7,8,434,3.52
8,5,318,2.58



--- TrafficType (tipo: int64) ---


,valor,contagem,percentagem
0,2,3913,31.74
1,1,2451,19.88
2,3,2052,16.64
3,4,1069,8.67
4,13,738,5.99
5,10,450,3.65
6,6,444,3.60
7,8,343,2.78
8,5,260,2.11
9,11,247,2.00



--- VisitorType (tipo: object) ---


,valor,contagem,percentagem
0,Returning_Visitor,10551,85.57
1,New_Visitor,1694,13.74
2,Other,85,0.69



--- Weekend (tipo: bool) ---


,valor,contagem,percentagem
0,False,9462,76.74
1,True,2868,23.26



--- Revenue (tipo: bool) ---


,valor,contagem,percentagem
0,False,10422,84.53
1,True,1908,15.47


ver se existem linhas duplicadas (exatamente iguais) e validar se não existem mesmo valores em falta

In [ ]:
# Verificar valores em falta
print("=== Missing Values ===")
missing = df.isnull().sum()
print(missing[missing > 0] if missing.sum() > 0 else "Não existem valores em falta.")

# Verificar duplicados
print("\n=== Duplicados ===")
n_duplicados = df.duplicated().sum()
print(f"Número de linhas duplicadas: {n_duplicados}")

=== Missing Values ===
Não existem valores em falta.

=== Duplicados ===
Número de linhas duplicadas: 125


In [8]:
# Identificar todas as linhas envolvidas em duplicação (originais + cópias)
duplicados = df[df.duplicated(keep=False)].copy()

# Criar um ID de grupo para cada conjunto de linhas idênticas
duplicados['grupo_id'] = duplicados.groupby(list(df.columns)).ngroup()

# Ordenar por grupo para ver os pares/conjuntos lado a lado
duplicados = duplicados.sort_values('grupo_id')

print(f"Total de linhas envolvidas em duplicação: {len(duplicados)}")
print(f"Número de grupos de duplicados distintos: {duplicados['grupo_id'].nunique()}")


Total de linhas envolvidas em duplicação: 201
Número de grupos de duplicados distintos: 76


201 linhas no total fazem parte de algum grupo de duplicados (isto inclui a "primeira ocorrência" de cada grupo + todas as cópias repetidas).
Essas 201 linhas distribuem-se por 76 grupos distintos, ou seja, há 76 combinações de valores diferentes que se repetem, cada uma com 2 ou mais linhas idênticas entre si.